katcp/resource.py

```diff
@@ -894,9 +892,7 @@ def handle_update(sensor, reading):
             # This handler is called whenever a sensor update is received
             try:
                 assert sensor is self
-                cond_matched = condition_test(reading)
-                status_matched = reading.status == status or status is None
```

Comment:
Yes `condition_test` does take entire reading and is free to check whatever but my point was that it does not:
```
882         condition_test = (condition_or_value if callable(condition_or_value)
                              else lambda s: s.value == condition_or_value)
```

But since you are the original author and happy with removing the status check, it does not really matter. Your reasoning makes sense to me.